core/src/main/kotlin/org/evomaster/core/problem/graphql/service/GraphqlSampler.kt

```diff
@@ -0,0 +1,131 @@
+    @Inject
+    private lateinit var configuration: EMConfig
+
+    private val adHocInitialIndividuals: MutableList<GraphqlAction> = mutableListOf()
+
+    private val modelCluster: MutableMap<String, ObjectGene> = mutableMapOf()
+
+    @PostConstruct
+    private fun initialize() {
+
+        GraphqlSampler.log.debug("Initializing {}", GraphqlSampler::class.simpleName)
+
+        rc.checkConnection()
+
+        val started = rc.startSUT()
+        if (!started) {
+            throw SutProblemException("Failed to start the system under test")
+        }
+
+        val infoDto = rc.getSutInfo()
+                ?: throw SutProblemException("Failed to retrieve the info about the system under test")
+
+        val schema = getSchema(infoDto)
+
+        actionCluster.clear()
+        modelCluster.clear()
+        // Here we should create Actions and models from schema
+
+    }
+
+    override fun sampleAtRandom(): GraphqlIndividual {
+        TODO("not implemented") //To change body of created functions use File | Settings | File Templates.
+    }
+
+    private fun getSchema(infoDto: SutInfoDto): TypeDefinitionRegistry
+    {
+
+        val graphqlEndpointUrl = infoDto?.graphqlProblem?.graphqlEndpointUrl ?: throw java.lang.IllegalStateException("Missing information about the graphql endpoint")
+
+        val query = IntrospectionQuery.INTROSPECTION_QUERY
+        val encodedQuery = URLEncoder.encode(query, "utf-8")
+        val response = connectToEndpoint(graphqlEndpointUrl, 30, encodedQuery)
+
+        if (!response.statusInfo.family.equals(Response.Status.Family.SUCCESSFUL)) {
+            throw SutProblemException("Cannot get graphql schema info from $graphqlEndpointUrl , status=${response.status}")
+        }
+
+
+        val json = response.readEntity(String::class.java)
+
+        var map = mapOf<String,Object>()
+        val gson = Gson()
+        map = gson.fromJson(json, object : TypeToken<Map<String, Object>>() {}.type)
+        var mapJson = gson.toJson(map.getValue("data"))
+
+        var schemaMap = mapOf<String,Object>()
+        schemaMap = gson.fromJson(mapJson, object : TypeToken<Map<String, Object>>() {}.type)
+        val document = IntrospectionResultToSchema().createSchemaDefinition(schemaMap)
+
+        val schema = try{
+            SchemaParser().buildRegistry(document)
+        } catch (e: Exception) {
+            throw SutProblemException("Failed to parse graphql schema: $e")
+        }
+
+        return schema
+
+    }
+
+    private fun connectToEndpoint(endPointUrl: String, attempts: Int, query: String): Response {
+
+        for (i in 0 until attempts) {
+            try {
+                return ClientBuilder.newClient()
+                        .target(endPointUrl + "?query=" + query)
+                        .request(MediaType.APPLICATION_JSON_TYPE)
+                        .get()
+            } catch (e: Exception) {
+
+                if (e.cause is ConnectException) {
+                    Thread.sleep(1_000)
```

Comment:
i don't think we should use such an approach of retries. for example, if the SUT is down, here we get stuck for 30 seconds...
the startSUT command should be blocking. if, once called, the SUT is not up and running, then there should be no point to try 30 times, isn't it?

Reply:
Yes you are right, but it's the same approach that you have used in Rest problem.
Is there a reason for using this approach in connecting to swagger endpoint ?

Reply:
fair point. didn't realize was the same code I wrote. however, in that I had the comment to clarify it: "  //Even if SUT is running, Swagger service might not be ready yet. So let's just wait a bit, and then retry"
re-using the same approach is fine. but we should avoid copy&paste of the same code (which should be refactored in some utility class, or at least marked with  a FIXME for later)

Reply:
Yes you are right, I'll fixed it in the next commit.